src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -455,36 +466,44 @@ public void testCacheFileChangeInConfig() throws Exception {
         final DefaultConfiguration checkerConfig = new DefaultConfiguration("configuration");
         checkerConfig.addAttribute("charset", "UTF-8");
         checkerConfig.addChild(treeWalkerConfig);
-        checkerConfig.addAttribute("cacheFile", temporaryFolder.newFile().getPath());
+        final File cacheFile = temporaryFolder.newFile();
+        checkerConfig.addAttribute("cacheFile", cacheFile.getPath());
 
         final Checker checker = new Checker();
-        final Locale locale = Locale.ROOT;
-        checker.setLocaleCountry(locale.getCountry());
-        checker.setLocaleLanguage(locale.getLanguage());
         checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
-        checker.configure(checkerConfig);
         checker.addListener(new BriefUtLogger(stream));
+        checker.configure(checkerConfig);
 
         final String pathToEmptyFile = temporaryFolder.newFile("file.java").getPath();
         final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
 
         verify(checker, pathToEmptyFile, pathToEmptyFile, expected);
+        final Properties cacheAfterFirstRun = new Properties();
+        cacheAfterFirstRun.load(Files.newBufferedReader(cacheFile.toPath()));
 
-        // update Checker config
+        // clean up the checker and persist the cache
         checker.destroy();
-        checker.configure(checkerConfig);
 
         final Checker otherChecker = new Checker();
-        otherChecker.setLocaleCountry(locale.getCountry());
-        otherChecker.setLocaleLanguage(locale.getLanguage());
         otherChecker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
-        otherChecker.configure(checkerConfig);
         otherChecker.addListener(new BriefUtLogger(stream));
         // here is diff with previous checker
         checkerConfig.addAttribute("fileExtensions", "java,javax");
+        otherChecker.configure(checkerConfig);
 
         // one more time on updated config
         verify(otherChecker, pathToEmptyFile, pathToEmptyFile, expected);
+        final Properties cacheAfterSecondRun = new Properties();
+        cacheAfterSecondRun.load(Files.newBufferedReader(cacheFile.toPath()));
+
+        assertEquals(
+            cacheAfterFirstRun.getProperty(pathToEmptyFile),
+            cacheAfterSecondRun.getProperty(pathToEmptyFile)
+        );
+        assertNotEquals(
+            cacheAfterFirstRun.getProperty("configuration*?"),
+            cacheAfterSecondRun.getProperty("configuration*?")
+        );
```

Comment:
Can we add an assertion that we have the same number of properties in the cache?